src/tests/functional/rendering_tests.js

```diff
@@ -440,6 +442,20 @@ test("test restores focus during turbo-frame rendering when transposing the acti
   assert.ok(await selectorHasFocus(page, "#permanent-input-in-frame"), "restores focus after page loads")
 })
 
+test("test restores focus during turbo-frame morphing when transposing the activeElement", async ({ page }) => {
+  const input = await page.locator("#permanent-input-in-frame")
+  const frame = await page.locator("turbo-frame#hello")
+
+  await frame.evaluate((frame) => frame.setAttribute("refresh", "morph"))
+  await input.press("Enter")
+  await nextEventOnTarget(page, "hello", "turbo:frame-render")
+
+  assert.ok(await locatorHasFocus(input), "restores focus after page loads")
+  assert.equal(await input.count(), 1)
+  assert.equal(await frame.count("#permanent-form-in-frame"), 1)
+  assert.equal(await frame.count("#permanent-descendant-input-in-frame"), 1)
```

Comment:
Something is off, I can't figure out a way to pass these. When the morph occurs, it duplicates the `[data-turbo-permanent]` elements.